src/presentation/controllers/device-controller.ts

```diff
@@ -0,0 +1,131 @@
+import { TelemetricsDevice, TelemetricsDeviceRestDTO } from '../../model/entities/telemetrics-device'
+
+@RestController('/devices')
+export class DeviceController extends BaseController{
+
+  constructor(
+    private readonly deviceService: DeviceService,
+    private readonly httpErrorHandler: HttpErrorHandler,
+    private readonly httpResponseFormatter: HttpResponseFormatter
+  ) {
+    super()
+  }
+
+  @GetRequest('/health')
+  public health(_request: Request, response: Response): void {
+    try {
+      const healthcheck = {
+        statusCode: 200,
+        state: 'Healthy'
+      }
+      response.send(this.httpResponseFormatter.formatSuccessResponse(new HealthDto(healthcheck.statusCode, healthcheck.state)))
+    } catch (error) {
+      this.httpErrorHandler.handleError(response, error as Exception)
+    }
+  }
+
+  @GetRequest('/')
+  public async getAllDeviceConfigurations(_request: Request, response: Response): Promise<void> {
+    try {
+      const configs = await this.deviceService.readAllConfigurations()
+      response.send(this.httpResponseFormatter.formatSuccessResponse(
+        configs.map(config => {
+          switch (config.type) {
+            case 'INewsDevice':
+              // eslint-disable-next-line no-case-declarations
+              const incfg = config as INewsDevice
+              return new INewsDeviceRestDTO(config, incfg.username, incfg.password)
+            case 'TelemetricsDevice': // Corrected case label
+              // eslint-disable-next-line no-case-declarations
+              const tmcfg = config as TelemetricsDevice
+              return new TelemetricsDeviceRestDTO(config, tmcfg.host)
+            default:
+              throw new Error(`Unknown device type: ${config.type}`)
+          }
+        })
+      ))
+    } catch (error) {
+      this.httpErrorHandler.handleError(response, error as Exception)
+    }
+  }
+
+
+  @GetRequest('/:id')
+  public async getDeviceConfiguration(request: Request, response: Response): Promise<void> {
+    try {
+      const deviceId = request.params.id
+      const config = await this.deviceService.readConfiguration(deviceId)
+      response.send(this.httpResponseFormatter.formatSuccessResponse(new DeviceDto(config)))
+    } catch (error) {
+      this.httpErrorHandler.handleError(response, error as Exception)
+    }
+  }
+
+  @PostRequest('/')
+  public async createDeviceConfiguration(request: Request, response: Response): Promise<void> {
+    try {
+      const deviceDto: DeviceRestDto = request.body as DeviceRestDto //discriminator field
+
+      switch(deviceDto.type){
+        case 'INewsDevice':
+          await this.deviceService.create(request.body as INewsDevice)
+          break
+        case 'TelemetricsDevice':
+          await this.deviceService.create(request.body as TelemetricsDevice)
+          break
+        default:
+          console.log('createDeviceConfiguration rececived a malformed request') //Help, replace me with a proper logger
```

Comment:
Can't you use the logger we have in the system already? :) The one that should sit in logger/logger.ts?

Reply:
Yes, I will. Just used console.log while rigging it all up. Will inject and use it now.